azure/scope/cluster.go

```diff
@@ -289,20 +290,24 @@ func (s *ClusterScope) NSGSpecs() []azure.NSGSpec {
 }
 
 // SubnetSpecs returns the subnets specs.
-func (s *ClusterScope) SubnetSpecs() []azure.SubnetSpec {
+func (s *ClusterScope) SubnetSpecs() []azure.ResourceSpecGetter {
 	numberOfSubnets := len(s.AzureCluster.Spec.NetworkSpec.Subnets)
 	if s.IsAzureBastionEnabled() {
 		numberOfSubnets++
 	}
 
-	subnetSpecs := make([]azure.SubnetSpec, 0, numberOfSubnets)
+	subnetSpecs := make([]azure.ResourceSpecGetter, 0, numberOfSubnets)
+
 	for _, subnet := range s.AzureCluster.Spec.NetworkSpec.Subnets {
-		subnetSpec := azure.SubnetSpec{
+		subnetSpec := &subnets.SubnetSpec{
 			Name:              subnet.Name,
+			ResourceGroup:     s.ResourceGroup(),
```

Comment:
I think we may also have to pass vnet's resource group because that is where subnet will be created : https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/a8555bbe56ce398efdd984906f059abef8e89d00/azure/services/subnets/subnets.go#L103

This is going to be tricky since we don't have a place in `azure.ResourceSpecGetter` to accomodate this. 
cc @CecileRobertMichon

Reply:
actually shouldn't `s.Scope.Vnet().ResourceGroup` be the only RG that gets passed in since that's the RG for the subnet resource? Why do we need ` s.ResourceGroup()` at all?

Reply:
I think it's needed to refer sub resource id's => https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/a8555bbe56ce398efdd984906f059abef8e89d00/azure/services/subnets/subnets.go#L90

Reply:
we could probably work around that by passing the id directly when creating the spec

Reply:
in any case `ResourceGroupName` should return the vnet RG for a subnet spec

Reply:
@shysank @CecileRobertMichon If the user is bringing a custom VNet then wouldn't the NAT gateways, route tables, and security groups be on the VNet's resource group since we're not creating them in the reconciler?